[COOK-2730] Allow specifying the proxy_pass domain manually incase the app name doesn't match

benlangfeld opened the pull request:
http://tickets.opscode.com/browse/COOK-2730

benlangfeld committed:
Allow setting host, X-Real-IP and X-Forwarded-For headers

josephholsten commented on templates/default/load_balancer.conf.erb:
```diff
@@ -18,6 +18,9 @@ server {
-    proxy_pass http://<%= @resource.application.name %>;
+    proxy_pass http://<%= @resource.proxy_pass || @resource.application.name %>;
+    <% if @resource.set_host_header %>
+    proxy_set_header Host $host;
```
when we use `proxy_set_header`, we also have the line:

```
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```

josephholsten commented:
I've created a ticket for you: http://tickets.opscode.com/browse/COOK-2730

sethvargo commented:
@benlangfeld there are some comments on the JIRA ticket that require your attention.

benlangfeld commented:
I've added some documentation for set_host_header, and reverted the proxy_pass change since it wasn't actually the appropriate solution to my problem.

Feel free to squash to a single commit. Anything else required?

sethvargo commented:
@benlangfeld this does not merge cleanly. Could you please rebase and mark the ticket as Fix Provided?

benlangfeld commented:
Rebased and squashed to a single commit.

someara commented:
merged at 7b32e59e8c6092d9d35f854a188e4367a90537d9
